### ISSUE TITLE
[android] disabling gestures while animation is in progress

### ISSUE THREAD

**Issue by LukasPaczos** [mapbox/mapbox-gl-native#9172]
**Platform:** Android
**Mapbox SDK version:** v5.1.0-beta.3

I'm trying to block gestures while animation is in progress, but unfortunatelly cannot achieve that in any way.
On v4.2.0 I had no problems with this.

![animation_queue_issue](https://cloud.githubusercontent.com/assets/16925074/26717187/bcbe448e-477c-11e7-8655-37175c947f63.gif)

```
    public void tilt(View view) {
        map.getUiSettings().setAllGesturesEnabled(false);
        Log.d("Map", "disabling gestures");
        CameraPosition position = map.getCameraPosition();
        CameraPosition.Builder builder = new CameraPosition.Builder(position);
        if (tilted) {
            tilted = false;
            builder.tilt(0);
        }
        else {
            builder.tilt(60);
            tilted = true;
        }

        map.animateCamera(CameraUpdateFactory.newCameraPosition(builder.build()), 1500, new MapboxMap.CancelableCallback() {
            @Override
            public void onCancel() {
                Log.d("Map", "animation - canceled - enabling gestures");
                map.getUiSettings().setAllGesturesEnabled(true);
            }

            @Override
            public void onFinish() {
                map.getUiSettings().setAllGesturesEnabled(true);
                Log.d("Map", "animation - finished - enabling gestures");
            }
        });
    }
```

```
Map: disabling gestures
Map: animation - finished - enabling gestures
Map: disabling gestures
Map: animation - finished - enabling gestures
Map: disabling gestures
Map: animation - canceled - enabling gestures
Map: scrolling
Map: scrolling
Map: scrolling
Map: scrolling
Map: scrolling
```

**Comment by tobrun** [comment]
Was able to reproduce the issue. Additional note is when disabling gestures up front and doing an animation, you can still interrupt the animation. This shouldn't happen.

**Comment by LukasPaczos** [comment]
@tobrun I found an issue. In `MapGestureDetector#onScroll()` condition `if (!trackingSettings.isScrollGestureCurrentlyEnabled())` is being checked too late and animation is being canceled inside `if (!scrollInProgress)` condition.

Simply moving
```
if (!trackingSettings.isScrollGestureCurrentlyEnabled()) {
        return false;
      }
```
to the top of `onScroll()` method fixes the issue.

**Comment by tobrun** [comment]
noticed the same myself ;)

**Comment by LukasPaczos** [comment]
@tobrun could you create a PR with this fix? I don't have a setup to do so unfortunately.

**Comment by tobrun** [comment]
closed with #9173
